docusaurus/docs/iOS/swiftui/swiftui-cookbook/creating-channels.md

```diff
@@ -0,0 +1,560 @@
+                    channelController: controller
+                )
+            } else {
+                Spacer()
+            }
+        }
+        .navigationTitle("New Chat")
+        .onReceive(keyboardWillChangePublisher) { visible in
+            keyboardShown = visible
+        }
+        .modifier(HideKeyboardOnTapGesture(shouldAdd: keyboardShown))
+    }
+}
+```
+
+The code above builds the UI for adding new channels. It consists of a text input view, where you can search for users available on the platform. When you select a user, it creates a channel view to start sending messages and shows the added users above the text input. 
+
+In the trailing part of the text input, there's a button to add additional users to the chat. 
+
+Below the text input view, there is a list that shows the available users to add to a chat.
+
+In the code above, we are also using some helper views, that help us achieve the desired user interface.
+
+```swift
+struct SelectedUserView: View {
+
+    @Injected(\.colors) var colors
+
+    var user: ChatUser
+
+    var body: some View {
+        HStack {
+            MessageAvatarView(
+                avatarURL: user.imageURL,
+                size: CGSize(width: 20, height: 20)
+            )
+
+            Text(user.name ?? user.id)
+                .lineLimit(1)
+                .padding(.vertical, 2)
+                .padding(.trailing)
+        }
+        .background(Color(colors.background1))
+        .cornerRadius(16)
+    }
+}
+
+struct SearchUsersView: View {
+
+    @StateObject var viewModel: NewChatViewModel
+
+    var body: some View {
+        HStack {
+            TextField("Type a name", text: $viewModel.searchText)
+            Button {
+                if viewModel.state == .channel {
+                    withAnimation {
+                        viewModel.state = .loaded
+                    }
+                }
+            } label: {
+                Image(systemName: "person.badge.plus")
+            }
+        }
+    }
+}
+
+struct VerticallyCenteredView<Content: View>: View {
+
+    var content: () -> Content
+
+    var body: some View {
+        VStack {
+            Spacer()
+            content()
+            Spacer()
+        }
+    }
+}
+
+struct CreateGroupButton: View {
+
+    @Injected(\.colors) var colors
+    @Injected(\.fonts) var fonts
+
+    @Binding var isNewChatShown: Bool
+
+    var body: some View {
+        NavigationLink {
+            CreateGroupView(isNewChatShown: $isNewChatShown)
+        } label: {
+            HStack {
+                Image(systemName: "person.3")
+                    .renderingMode(.template)
+                    .foregroundColor(colors.tintColor)
+
+                Text("Create a group")
+                    .font(fonts.bodyBold)
+                    .foregroundColor(Color(colors.text))
+
+                Spacer()
+            }
+            .padding()
+        }
+        .isDetailLink(false)
```

Comment:
You sure this should be always `false`? 🤔

```suggestion
        .isDetailLink(UIDevice.current.userInterfaceIdiom == .pad)
```

Reply:
Yes, false is fine for here.